immuneML/ml_methods/GenerativeModel.py

```diff
@@ -0,0 +1,185 @@
+import abc
+import os
+import warnings
+from pathlib import Path
+
+import dill
+import pkg_resources
+import yaml
+from sklearn.utils.validation import check_is_fitted
+
+from immuneML.data_model.dataset.Dataset import Dataset
+from immuneML.data_model.encoded_data.EncodedData import EncodedData
+from immuneML.environment.Label import Label
+from immuneML.ml_methods.UnsupervisedMLMethod import UnsupervisedMLMethod
+from immuneML.util.FilenameHandler import FilenameHandler
+from immuneML.util.PathBuilder import PathBuilder
+
+
+class GenerativeModel(UnsupervisedMLMethod):
+
+    def __init__(self, parameter_grid: dict = None, parameters: dict = None):
+        super(GenerativeModel, self).__init__()
+        self.model = None
+
+        if parameter_grid is not None and "show_warnings" in parameter_grid:
+            self.show_warnings = parameter_grid.pop("show_warnings")[0]
+        elif parameters is not None and "show_warnings" in parameters:
+            self.show_warnings = parameters.pop("show_warnings")
+        else:
+            self.show_warnings = True
+
+        self._parameter_grid = parameter_grid
+        self._parameters = parameters
+        self._length_of_sequence = 20 #Set as average, will be overwritten
+        self.feature_names = None
+        self.class_mapping = None
+        self.label = None
+
+    def fit(self, encoded_data, cores_for_training: int = 2, result_path: Path = None):
+        self._length_of_sequence = 21# if "length_of_sequence" not in encoded_data.info else encoded_data.info["length_of_sequence"]
+        self.model = self._fit(encoded_data.examples, cores_for_training, result_path)
+
+    def generate(self, amount=10, path_to_model: Path = None):
+        pass
+
+    def _fit(self, X, cores_for_training: int = 1):
+
+        self.model = self._get_ml_model(X)
```

Comment:
better to leave it abstract than have it do the wrong thing even though it is overridden by subclass implementation :)